UnityProject/Assets/Scripts/Objects/Engineering/APC.cs

```diff
@@ -161,29 +230,86 @@ private void UpdateDisplay()
 
 		/// <summary>
 		/// Change brightness of lights connected to this APC proportionally to voltage
+		/// additionally runs integrity and over-drawing checks. Over-drawing your APC can
+		/// result in !!FUN!! things happening, while letting your integrity drop too low
+		/// can cause devices to disconnect randomly. (Proportinate to your integrity)
 		/// </summary>
 		private void HandleDevices()
 		{
 			float Voltages = Voltage;
 			if (Voltages > 270)
 			{
+				integrity.ApplyDamage(25, AttackType.Melee, DamageType.Burn);
 				Voltages = 0.001f;
+			} else if (Voltages > 250)
+			{
+				integrity.ApplyDamage(5, AttackType.Melee, DamageType.Burn);
 			}
 			float CalculatingResistance = new float();
 
+			if (integrity.integrity <= (integrity.initialIntegrity *0.10))
```

Comment:
this looks like duplicated code from OnDamageReceived. Extract this code into its own method and call that instead.

Reply:
You're also applying damage above, so the integrity check will happen twice.